src/RequestClientWrapper.ts

```diff
@@ -97,21 +100,26 @@ export default class RequestClientWrapper {
     );
   }
 
-  private makeRequest(
+  private async makeRequest(
     method: HttpRequestMethod,
     callStart: Date,
     numberOfCalls: number,
     url: string,
     options: AxiosRequestConfig,
     callback: RequestCallback
-  ): void {
+  ): Promise<void> {
     const callDetails: CallDetails = { options, callStart, callCountForThisRequest: numberOfCalls };
     options.baseURL = this.requestClientWrapperConfiguration.endpoint;
     if (options.headers === undefined) {
       options.headers = {};
     }
-    options.headers['X-Channel-Ape-Authorization-Token'] = this.requestClientWrapperConfiguration.session;
-    options.headers['Content-Type'] = 'application/json';
+    if (this.isItJwtToken(this.requestClientWrapperConfiguration.session)) {
+      options.headers['X-Channel-Ape-Jwt'] = await this.refreshSupabaseSession();
+      options.headers['Content-Type'] = 'application/json';
```

Comment:
I think you can move the 'Content-Type' header outside if/else and only set one time. Both your if/else are setting it here.